pkg/platform/runtime/setup/setup.go

```diff
@@ -247,27 +296,31 @@ func (s *Setup) deleteOutdatedArtifacts(changeset artifact.ArtifactChangeset, st
 	return nil
 }
 
-func (s *Setup) installFromBuildResult(buildResult *model.BuildResult, artifacts map[artifact.ArtifactID]artifact.ArtifactRecipe) error {
+// setupArtifactSubmitFunction returns a function that sets up an artifact and can be submitted to a workerpool
+func (s *Setup) setupArtifactSubmitFunction(a artifact.ArtifactDownload, buildResult *model.BuildResult, artifacts artifact.ArtifactRecipeMap, errors []error, secondStageCh chan<- func() error) func() {
+	return func() {
+		// 'bundle' makes sense for the single camel download
```

Comment:
This feels wrong, why are we making an assumption about camel downloads outside of the camel package?

Reply:
Well, I guess it also makes sense for `alternative` artifacts, but that of course shouldn't happen.  We should always be able to resolve that to a "real" name.

It is just that for camel builds, there is no such name, as all artifacts are combined in one bundle.  I'll try to improve the comment.